src/main/java/org/opensearch/sdk/handlers/ClusterStateResponseHandler.java

```diff
@@ -68,11 +68,14 @@ public ClusterStateResponse read(StreamInput in) throws IOException {
 
     /**
      * Invokes await on the ClusterStateResponseHandler count down latch
-     * @throws InterruptedException
+     * @throws Exception
      *     if the response times out
      */
-    public void awaitResponse() throws InterruptedException {
-        inProgressLatch.await(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+    public void awaitResponse() throws Exception {
+        inProgressFuture.orTimeout(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+        if (inProgressFuture.isCompletedExceptionally()) {
```

Comment:
Just for my understanding why are we checking this condition? Wouldn't `get` take care of it?

Reply:
I think we actually are going to change this, pending CR of [#5646](https://github.com/opensearch-project/OpenSearch/pull/5646).   

TLDR: we need to use either `.get()` or `.join()` in all cases or it won't behave as we expect.

Reply:
Please change this to:
```java
inProgressFuture.orTimeout(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).get();
```
and remove the next 3 lines.